### PR TITLE
Move component tests from /tests to component directories.

### LINKED ISSUE #1690: Move tests to correct component
As disscussed in https://github.com/sulu-io/sulu/issues/1679#issuecomment-147744005 it would be good to move tests into the corresponding compontents.

### PR DESCRIPTION
**Pull request by dantleech**
This PR moves all of the component tests from `/tests` to `src/Sulu/Components/*/Tests`.

Things to consider:

- [x] Organization of Functional and Unit tests (components do have both).
- [x] Autoloader optimizationg, e.g. `exclude_from_classmap`, see [symfony composer.json](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/composer.json)

NOTE: This PR does not currently do the following things which may be done a followup PR:

- Move function functional test setup from `/tests/app` and `/tests/Resources`

Fixes #1690

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -2,17 +2,22 @@
 <phpunit bootstrap="./tests/bootstrap.php" colors="true">
 
     <testsuites>
-        <testsuite name="Sulu Test Suite">
-            <directory suffix="Test.php">./tests</directory>
+        <testsuite name="Sulu Unit tests">
+            <directory suffix="Test.php">./src/Sulu/Component/*/Tests/Unit</directory>
+        </testsuite>
+        <testsuite name="Sulu Functional tests">
```
Comment (dantleech) [inline phpunit.xml.dist]: remove ` tests`. it is self-evident.

File: src/Sulu/Component/Webspace/Tests/Unit/WebspaceTestCase.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Webspace\Tests\Unit;
+
+class WebspaceTestCase extends \PHPUnit_Framework_TestCase
+{
+    protected function getResourceDirectory()
+    {
+        $dir = __DIR__ . '/../../../../../../tests/Resources';
```
Comment (dantleech) [inline src/Sulu/Component/Webspace/Tests/Unit/WebspaceTestCase.php]: Sharing this resource directory is obviously bad -- but removing it is another PR as it affects the bundle tests too.

**Comment by danrot** [comment]
Changelog is missing. And the autoloading isn't optimized yet, right? Shouldn't that be a very easy task, which can be added here?

**Comment by dantleech** [comment]
Yeah right. Need to experiment with the autoloader optimization.

**Comment by dantleech** [comment]
Please review. If it is OK I will rebase and merge. @danrot

**Pull request merged**